components/raftstore/src/store/peer.rs

```diff
@@ -2020,6 +2029,7 @@ where
                     // prewrites or commits will be just a waste.
                     self.last_urgent_proposal_idx = self.raft_group.raft.raft_log.last_index();
                     self.raft_group.skip_bcast_commit(false);
+                    self.last_sent_snapshot_idx = self.raft_group.raft.raft_log.last_index();
```

Comment:
Why reset `snapshot_idx` when the role changed to leader?

Reply:
Because new leader has no idea what's the snapshot index sent by last leader.